### PR TITLE
[CARBONDATA-3265] Fixed memory leak in Range Sort

### PR DESCRIPTION
**Pull request by shivamasn**
In range sort, unsafe memory was not getting cleared in case of task failure . So, added a fix for memory leak.

Be sure to do all of the following checklist to help us incorporate 
your contribution quickly and easily:

 - [ ] Any interfaces changed?
 
 - [ ] Any backward compatibility impacted?
 
 - [ ] Document update required?

 - [ ] Testing done
        Please provide details on 
        - Whether new unit test cases have been added or why no new tests are required?
        - How it is tested? Please attach test report.
        - Is it a performance related change? Please attach the performance test report.
        - Any additional information to help reviewers in testing this change.
       
 - [ ] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA.

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/2398/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/2624/

File: integration/spark-common/src/main/scala/org/apache/carbondata/spark/load/DataLoadProcessBuilderOnSpark.scala [hunk 1]
```diff
@@ -221,9 +247,32 @@ object DataLoadProcessBuilderOnSpark {
...
+        CommonUtil.clearUnsafeMemory(ThreadLocalTaskInfo.getCarbonTaskInfo.getTaskId)
+      }
+      ThreadLocalSessionInfo.setCarbonSessionInfo(carbonSessionInfo)
+      TaskMetricsMap.threadLocal.set(Thread.currentThread().getId)
+      val carbonTaskInfo = new CarbonTaskInfo
+      carbonTaskInfo.setTaskId(CarbonUtil.generateUUID())
+      ThreadLocalTaskInfo.setCarbonTaskInfo(carbonTaskInfo)
+      carbonSessionInfo.getSessionParams.getAddedProps.asScala.map {
+        f => CarbonProperties.getInstance().addProperty(f._1, f._2)
+      }
```
Comment (qiuchenjian) [inline integration/spark-common/src/main/scala/org/apache/carbondata/spark/load/DataLoadProcessBuilderOnSpark.scala]: seems that two pieces of code are the same， better to use a method

File: integration/spark-common/src/main/scala/org/apache/carbondata/spark/load/DataLoadProcessBuilderOnSpark.scala [hunk 2]
```diff
@@ -127,9 +130,32 @@ object DataLoadProcessBuilderOnSpark {
       }
 
     // 4. Write
-    sc.runJob(sortRDD, (context: TaskContext, rows: Iterator[CarbonRow]) =>
+    sc.runJob(sortRDD, (context: TaskContext, rows: Iterator[CarbonRow]) => {
+      TaskContext.get.addTaskCompletionListener(_ => ThreadLocalSessionInfo.unsetAll())
+      val carbonSessionInfo: CarbonSessionInfo = {
+        var info = ThreadLocalSessionInfo.getCarbonSessionInfo
+        if (info == null || null == info.getSessionParams) {
```
Comment (qiuchenjian) [inline integration/spark-common/src/main/scala/org/apache/carbondata/spark/load/DataLoadProcessBuilderOnSpark.scala]: if info ！= null ，no need to generate a new object

**Comment by QiangCai** [comment]
please modify the title to use range column instead of Range Partition

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/2425/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.3.2, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/10683/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/2652/

File: integration/spark-common/src/main/scala/org/apache/carbondata/spark/load/DataLoadProcessBuilderOnSpark.scala (hunk 2 above)
Comment (shivamasn) [reply integration/spark-common/src/main/scala/org/apache/carbondata/spark/load/DataLoadProcessBuilderOnSpark.scala]: This piece of code is not required, so I have removed it.

**Comment by kunal642** [comment]
LGTM

**Pull request closed without merging**